kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -227,9 +259,51 @@ object Channel:
       * @warning
       *   The actual capacity may be larger than the specified capacity due to rounding.
       */
-    def init[A](capacity: Int, access: Access = Access.MultiProducerMultiConsumer)(using Frame): Channel[A] < IO =
+    def init[A](capacity: Int, access: Access = Access.MultiProducerMultiConsumer)(using PutType[A], Frame): Channel[A] < IO =
         IO.Unsafe(Unsafe.init(capacity, access))
 
+    sealed trait PutType[A]:
```

Comment:
This is included to deal with the issue of `puts` being either a single value or a chunk of values. It's impossible to check whether `A | Chunk[A]` is a single value or a chunk in the generic case, because `A` might itself be a chunk. On the other hand, I want to avoid wrapping them if we don't need to, as this will add an allocation. The `PutType` trait provides ad-hoc polymorphism, so that the put type will be `A | Chunk[A]` for any `Channel[A]` where `A` is not a `Chunk` of something, and the put type will be `Wr[Chunk[A]] | Chunk[Chunk[A]]` for any `Channel[Chunk[A]]`. (`Wr` is just a simply wrapping case class that allows safely pattern matching.)

Reply:
(I'm not sure if all this is worth the trouble. Maybe there's a simpler approach?)